packages/app/contexts/StrategyContext.tsx

```diff
@@ -0,0 +1,65 @@
+"use client";
+
+import { FREQUENCY_OPTIONS } from "@/models/stack";
+import { Token } from "@/models/token";
+import { createContext, useContext, ReactNode, useMemo, useState } from "react";
+
+export type Strategy = {
+  id: number;
+  buyToken: Token;
+  daysAmount: number;
+  frequency: FREQUENCY_OPTIONS;
+  sellAmountPerTimeframe: number;
+  sellToken: Token;
+  totalSellAmount: string;
+};
+
+const throwStrategyContextError = () => {
+  throw new Error("No StrategyContext available");
+};
+
+interface StrategyContextProps {
+  selectedStrategy: Strategy | null;
```

Comment:
I think what we want here is for the type to handle:`undefined`

```suggestion
  selectedStrategy?: Strategy;
```